packages/slate-react/src/components/leaf.tsx

```diff
@@ -40,7 +40,7 @@ const Leaf = (props: {
     return () => {
       editorEl.style.minHeight = 'auto'
     }
-  }, [placeholderRef])
+  }, [placeholderRef, leaf])
```

Comment:
Adding this dependency actually executes this hook now when the placeholderRef has a element referenced. This didn't work before